src/screens/groups/batch-payment-screen.tsx

```diff
@@ -0,0 +1,251 @@
+}
+
+const Screen = ({ groupId }: Props) => {
+  const isMobile = useIsMobile()
+  const { connected } = useContext(Web3Context)
+
+  const [data, setData] = useState<GroupWithRecipients | undefined>(undefined)
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string>('')
+
+  const [selectedRecipients, setSelectedRecipients] =
+    useState<SelectedRecipients>({})
+  const [transaction, setTransaction] = useState<string>('')
+  const [paymentTransactionData, setPaymentTransactionData] =
+    useState<PaymentTransactionData | null>(null)
+
+  const [transactionDialogOpen, setTransactionDialogOpen] =
+    useState<boolean>(false)
+  const [transactionInfoDialogOpen, setTransactionInfoDialogOpen] =
+    useState<boolean>(false)
+
+  const handleRowSelect = (id: string) => {
+    setSelectedRecipients((prevState) => ({
+      ...prevState,
+      [id]: {
+        selected: !prevState[id]?.selected,
+        amount: prevState[id]?.amount,
+        address: prevState[id]?.address,
+      },
+    }))
+  }
+
+  const handleSelectAll = () => {
+    const allIds = data?.recipients.map((recipient) => recipient.id)
+    const areAllSelected = allIds?.every(
+      (id) => selectedRecipients[id]?.selected
+    )
+
+    const newSelectedRecipients: SelectedRecipients = {}
+
+    allIds?.forEach((id) => {
+      newSelectedRecipients[id] = {
+        selected: !areAllSelected,
+        amount: selectedRecipients[id]?.amount,
+        address: selectedRecipients[id]?.address,
+      }
+    })
+
+    setSelectedRecipients(newSelectedRecipients)
+  }
+
+  const handleAmountChange = (id: string, amount: number) => {
+    setSelectedRecipients((prevState) => ({
+      ...prevState,
+      [id]: {
+        selected: prevState[id]?.selected,
+        amount: amount,
+        address: prevState[id]?.address,
+      },
+    }))
+  }
+  const isSelectAllChecked =
+    Object.values(selectedRecipients).filter((recipient) => recipient.selected)
+      .length === data?.recipients.length
+
+  const onSendPaymentClicked = () => {
+    const recipients: string[] = []
+    const amounts: number[] = []
+
+    let totalAmount = 0
+    Object.entries(selectedRecipients).forEach(([, recipient]) => {
+      if (recipient.selected && recipient.amount > 0) {
+        recipients.push(recipient.address)
+        amounts.push(recipient.amount)
+        totalAmount += recipient.amount
+      }
+    })
+
+    setPaymentTransactionData({
+      recipients: recipients,
+      amounts: amounts,
+      totalAmount: totalAmount,
+    })
+    setTransactionDialogOpen(true)
+  }
+
+  const onTransactionSaved = (tx: string) => {
+    setPaymentTransactionData(null)
+    setTransaction(tx)
+    setTransactionInfoDialogOpen(true)
+  }
+
+  useEffect(() => {
+    if (!groupId) return
+
+    const fetchData = async () => {
+      setIsLoading(true)
+
+      try {
+        const group = await api.getGroupById(groupId)
```

Comment:
We should not list recipients without wallets.

There are two possible ways how to optimize such code:

1. Implement new function on API side to return only recipients with non-empty wallet
2. Filterize them on frontend side and keep `getGroupById` as it is right now

WDYT?

Reply:
For me 2 option is more easy to implement, but option 2 provides us a way to test such an API endpoint.

Reply:
It's not necessary right now, maybe we can move this discussion and create a new issue.

Reply:
option 1 looks good to me. and yes, it's good to create a new issue for this

Reply:
Let's create an issue 👍